lnst/Tests/XDPBench.py

```diff
@@ -0,0 +1,139 @@
+import re
+import time
+import signal
+import logging
+from subprocess import Popen, PIPE
+from lnst.Devices.Device import Device
+
+from lnst.Tests.BaseTestModule import BaseTestModule, TestModuleError
+from lnst.Common.Parameters import (
+    ChoiceParam,
+    StrParam,
+    IntParam,
+    DeviceParam,
+)
+
+
+class XDPBenchOutputParser:
+    def __init__(self, process: Popen, output: list) -> None:
+        self._process = process
+        self._out = output
+
+    def parse_output(self):
+        stdout, stderr = self._process.communicate()
+
+        logging.debug("Stderr of xdp-bench:")
+        logging.debug(str(stderr))
+
+        self._parse_raw_results(stdout)
+
+        if not self._out:
+            raise TestModuleError(f"Could not parse xdp-bench output: {stdout}")
```

Comment:
Exactly, this method does not return anything. Since (almost) everything in python is stored as a pointer/reference, it doesn't need to.

`self._out` is just a reference to `XDPBench._res_data`, see https://github.com/LNST-project/lnst/pull/310/files#diff-fab965ef6bb50e158a3f0486357589d9be87461c9424e4efef2212b705108213R107

But I'll rewrite it, this might be kinda confusing

Reply:
ah... so the TestModule provides its own `res_data` into the parser and the parser fills it out... yeah that's a confusing way to do that i'd say.